### PR TITLE
[link] style: new modal design

### PR DESCRIPTION
**Pull request by tamrat-bay**

### CODE REVIEW

File: packages/editor-common/web/statics/styles/link-action-buttons.scss [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    padding: 12px 0;
+    width: 100%;
+  }
+}
+
+.actionButtons_FooterActions,
+.actionButtons_RemoveContainer {
+  display: flex;
+  button {
+    color: $settings-action-color-fallback;
```
Comment (tamrat-bay) [inline packages/editor-common/web/statics/styles/link-action-buttons.scss:39]: Check if needed

File: packages/editor-common/web/statics/styles/link-action-buttons.scss [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  }
+}
+
+.actionButtons_saveOnlyBtn {
+  height: 40px;
```
Comment (tamrat-bay) [inline packages/editor-common/web/statics/styles/link-action-buttons.scss:44]: pass as prop btnSize

File: packages/editor/web/src/RichContentEditor/InnerModal.jsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ class InnerModal extends Component {
       bottom: bottomAfterOverflowCheck || 'auto',
       border: 'solid 1px #ededed',
       background: 'rgb(255, 255, 255)',
-      borderRadius: '6px',
+      borderRadius: '0',
```
Comment (tamrat-bay) [inline packages/editor/web/src/RichContentEditor/InnerModal.jsx:37]: delete

File: packages/plugin-link/web/src/toolbar/TextLinkButton.jsx [hunk 4]
```diff
@@ -59,9 +59,9 @@ export default class TextLinkButton extends Component {
               maxWidth: 512,
               height: 390,
               border: '1px solid rgb(237, 237, 237)',
-              borderRadius: '6px',
+              // borderRadius: '6px',
```
Comment (tamrat-bay) [inline packages/plugin-link/web/src/toolbar/TextLinkButton.jsx:62]: delete

File: packages/editor-common/web/statics/styles/link-panel.scss [hunk 5]
```diff
@@ -106,22 +131,16 @@
 }
 
 .linkPanel_actionsDivider {
-  margin: 0 4px;
+  position: absolute;
+  width: 100%;
+  left: 0;
   border-bottom: 1px solid rgba(51, 51, 51, 0.1);
+  &.linkPanel_actionsDivider_mobile {
+    top: 56px;
```
Comment (tamrat-bay) [inline packages/editor-common/web/statics/styles/link-panel.scss:139]: try to use margin/padding

File: packages/editor-common/web/statics/styles/link-action-buttons.scss [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  width: 94px;
+  border: none;
+}
+
+.actionButtons_mobile_title_buttons_wrapper {
```
Comment (tamrat-bay) [inline packages/editor-common/web/statics/styles/link-action-buttons.scss:49]: better name

File: packages/editor-common/web/statics/styles/link-action-buttons.scss [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+.actionButtons_FooterActions,
+.actionButtons_RemoveContainer {
+  display: flex;
+  button {
+    color: $settings-action-color-fallback;
+  }
+}
+
+.actionButtons_saveOnlyBtn {
```
Comment (tamrat-bay) [inline packages/editor-common/web/statics/styles/link-action-buttons.scss:43]: better name

File: packages/editor-common/web/statics/locale/messages_en.json [hunk 8]
```diff
@@ -59,7 +59,7 @@
   "LinkPreview_RemovePreview_Tooltip": "Remove preview",
   "LinkPreview_Settings_Tooltip": "Link settings",
   "LineSpacingButton_Tooltip": "Line spacing",
-  "MobileLinkModal_Title": "Add a link",
```
Comment (tamrat-bay) [inline packages/editor-common/web/statics/locale/messages_en.json:62]: update content

File: packages/editor-common/web/src/Components/LinkComponents/LinkActionsButtons.jsx [hunk 9]
```diff
@@ -23,77 +33,66 @@ class LinkActionsButtons extends PureComponent {
       basicLinkPanel,
       hideUrlInput,
       isMobile,
+      saveBtnOnly,
```
Comment (tamrat-bay) [inline packages/editor-common/web/src/Components/LinkComponents/LinkActionsButtons.jsx:36]: better name

File: packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx [hunk 10]
```diff
@@ -40,30 +25,44 @@ class BasicLinkPanel extends PureComponent {
...
+            hasCheckboxes={hasCheckboxes}
             {...sharedPanelsProps}
           />
-          <div className={styles.linkPanel_actionsDivider} role="separator" />
         </div>
-        <LinkActionsButtons basicLinkPanel {...buttonsProps} />
+        {showSeparator && (
+          <div
+            className={classNames(styles.linkPanel_actionsDivider, {
+              [styles.linkPanel_actionsDivider_mobile]: isMobile,
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/BasicLinkPanel.jsx:60]: maybe move to css 
.linkPanel_actionsDivider {
@include phone {}
}

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx [hunk 11]
```diff
@@ -85,10 +84,10 @@ class LinkPanel extends Component {
 
   getTextInput() {
     return (
-      <input
-        ref={this.textInput}
+      <TextInput
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:87]: use createRef or move func outside

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx [hunk 12]
```diff
@@ -120,15 +119,27 @@ class LinkPanel extends Component {
...
+        })}
+        {...ariaProps}
+        role="form"
+      >
         {!hideUrlInput && (
           /* eslint-disable-next-line jsx-a11y/no-static-element-interactions */
-          <div className={styles.linkPanel_Input} onKeyDown={this.handleKeyDown}>
+          <div
+            className={classNames(styles.linkPanel_Input, {
+              [styles.basicPanel]: !hasCheckboxes,
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanel.jsx:139]: maybe try to move to css

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx [hunk 13]
```diff
@@ -84,20 +85,31 @@ export class LinkPanelDropdown extends Component {
...
     const { itemToString, formatMenuItem, itemHeight, textInputProps, value } = this.props;
-    const { selectedItem, items, fallbackChanged } = this.state;
+    const { selectedItem, items } = this.state;
     return (
       <Suspense
         fallback={
-          <Input
+          <TextInput
             {...textInputProps}
+            inputRef={ref => (this.textInput = ref)}
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx:107]: createRef

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx [hunk 14]
```diff
@@ -84,20 +85,31 @@ export class LinkPanelDropdown extends Component {
...
             value={value}
-            onChange={e => {
-              this.props.onChange(e.target.value);
+            onChange={value => {
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx:109]: move to this.onChange function

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx [hunk 15]
```diff
@@ -115,34 +127,43 @@ export class LinkPanelDropdown extends Component {
...
+            const textInputProps = getInputProps({
+              ...textInputProps,
+            });
+            return (
+              <div>
+                {/*<label {...getLabelProps()}>Enter a fruit</label>*/}
+                <TextInput
+                  {...getInputProps({ ...textInputProps })}
+                  onChange={value => textInputProps.onChange({ target: { value } })}
+                  inputRef={ref => (this.textInput = ref)}
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx:140]: createRef

File: packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx [hunk 16]
```diff
@@ -115,34 +127,43 @@ export class LinkPanelDropdown extends Component {
...
+                />
+                {(isOpen || this.props.isOpen) && List && (
+                  <Suspense fallback={<div>Loading...</div>}>
+                    <List
+                      className={styles.linkPanel_dropdownList}
+                      style={{ borderTop: '0', position: 'absolute' }}
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/LinkPanelDropdown.jsx:146]: move to css

File: packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanel.jsx [hunk 17]
```diff
@@ -42,40 +41,42 @@ class MultiSelectLinkPanel extends PureComponent {
...
-        <LinkActionsButtons {...buttonsProps} />
-        <div className={styles.multiSelectLinkPanel_header_mobile}>
-          <LinkIcon className={styles.multiSelectLinkPanel_mobileHeaderIcon} />
-          <div>{t('LinkTo_Modal_Header')}</div>
-        </div>
-
-        {this.renderMobileTabs()}
-      </>
+      <div>
+        <LinkActionsButtons {...this.props.buttonsProps} />
```
Comment (danarish4) [inline packages/editor-common/web/src/Components/LinkComponents/MultiSelectLinkPanel.jsx:72]: move buttonsProps to 2 lines up

**Review by danarish4 — APPROVED** [review]

**Review by danarish4 — APPROVED** [review]

**Pull request merged**